feature/gribi/otg_tests/hierarchical_weight_resolution_test/hierarchical_weight_resolution_test.go

```diff
@@ -0,0 +1,739 @@
+	ipv4FlowIP        = "203.0.113.0"
+	innerSrcIPv4Start = "198.18.0.0"
+	innerDstIPv4Start = "198.19.0.0"
+	ipv4PrefixLen     = 30
+	ipv4FlowCount     = 65000
+	nhEntryIP1        = "192.0.2.111"
+	nhEntryIP2        = "192.0.2.222"
+	nonDefaultVRF     = "VRF-1"
+	policyName        = "redirect-to-VRF1"
+	ipipProtocol      = 4
+)
+
+var (
+	dutPort1 = attributes{
+		Attributes: attrs.Attributes{
+			Desc:    "dutPort1",
+			Name:    "port1",
+			IPv4:    dutPort1IPv4(0),
+			IPv4Len: ipv4PrefixLen,
+		},
+		numSubIntf: 0,
+		ip:         dutPort1IPv4,
+	}
+
+	atePort1 = attributes{
+		Attributes: attrs.Attributes{
+			Name:    "port1",
+			MAC:     "02:00:01:01:01:01",
+			IPv4:    atePort1IPv4(0),
+			IPv4Len: ipv4PrefixLen,
+		},
+		numSubIntf: 0,
+		ip:         atePort1IPv4,
+		gateway:    dutPort1IPv4,
+	}
+
+	dutPort2 = attributes{
+		Attributes: attrs.Attributes{
+			Desc:    "dutPort2",
+			Name:    "port2",
+			IPv4:    dutPort2IPv4(0),
+			IPv4Len: ipv4PrefixLen,
+		},
+		numSubIntf: 18,
+		ip:         dutPort2IPv4,
+	}
+
+	atePort2 = attributes{
+		Attributes: attrs.Attributes{
+			Name:    "port2",
+			MAC:     "02:00:02:01:01:01",
+			IPv4:    atePort2IPv4(0),
+			IPv4Len: ipv4PrefixLen,
+		},
+		numSubIntf: 18,
+		ip:         atePort2IPv4,
+		gateway:    dutPort2IPv4,
+	}
+
+	// nhgIPv4EntryMap maps NextHopGroups to the ipv4 entries pointing to that NextHopGroup.
+	nhgIPv4EntryMap = map[uint64]string{
+		1: ipv4EntryPrefix,
+		2: cidr(nhEntryIP1, 32),
+		3: cidr(nhEntryIP2, 32),
+	}
+	// 'tolerance' is the maximum difference that is allowed between the observed
+	// traffic distribution and the required traffic distribution.
+	tolerance = 0.2
+)
+
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
+
+// dutPort1IPv4 returns ip address 192.0.2.1, for every vlanID.
+func dutPort1IPv4(uint8) string {
+	return "192.0.2.1"
+}
+
+// atePort1IPv4 returns ip address 192.0.2.2, for every vlanID
+func atePort1IPv4(uint8) string {
+	return "192.0.2.2"
+}
+
+// dutPort2IPv4 returns ip addresses starting 192.0.2.5, increasing by 4
+// for every vlanID.
+func dutPort2IPv4(vlan uint8) string {
+	return fmt.Sprintf("192.0.2.%d", vlan*4+5)
+}
+
+// atePort2IPv4 returns ip addresses starting 192.0.2.6, increasing by 4
+// for every vlanID.
+func atePort2IPv4(vlan uint8) string {
+	return fmt.Sprintf("192.0.2.%d", vlan*4+6)
+}
+
+// cidr taks as input the IPv4 address and the Mask and returns the IP string in
+// CIDR notation.
+func cidr(ipv4 string, ones int) string {
+	return ipv4 + "/" + strconv.Itoa(ones)
+}
+
+// filterPacketReceived uses ATE:EgressTracking bucket counters to create a map
+// with bucket-label as the Key and the percentage of packets-received for that
+// bucket as the Value.
+func filterPacketReceived(t *testing.T, flow string, ate *ondatra.ATEDevice) map[string]float64 {
+	t.Helper()
+
+	// Check the egress packets
+	path := gnmi.OTG().Flow(flow).TaggedMetricAny()
+	vlanTags := gnmi.GetAll(t, ate.OTG(), path.State())
```

Comment:
`path` is only used in the `gnmi.GetAll` for `vlanTags`. I don't think it's adding any clarity, so we should remove it just use the full path directly in `gnmi.GetAll`

Reply:
done